### DISCUSSION TITLE
[Plugin] Provide file relative/absolute path in context

### DISCUSSION THREAD

**Discussion by g-plane** [swc-project/swc#6699] [Ideas]
### Describe the feature

Currently there're only 3 items provided in plugin context:

- file name
- env
- cwd

I'd propose to add a new context information: file path. This can be relative path or absolute path. I'm creating a plugin that will inject some debug info which requires file path, but SWC can't provide this to plugin now.

**Comment by kwonoj** [comment]
https://github.com/swc-project/swc/blob/61245739eaf05be083c4b53d34e026cf2de5635a/crates/swc_common/src/plugin/metadata.rs#L55-L57

Filename supposed to include paths if possible. In my opinion if that doesn't work for some cases it needs to be correctly extended instead of trying to add new context.

**Reply by g-plane** [reply]
It doesn't work for me, because it's just the file name, not path. How can I get the path?

**Reply by kdy1** [reply]
It passes the `filename` provided by the user. Can you try using full path for `transformFile` or passing a full path to file as `filename` to `transform`?

**Reply by g-plane** [reply]
I'm just using `@swc/cli`, not via API. How can I pass it?

**Reply by ousiri** [reply]
How to access this metadata inside TransformVisitor for swc plugin?

**Comment by coder-xiaotian** [comment]
you can fork my repository, I used get_context to get the file path in my code. [https://github.com/coder-xiaotian/swc-useclient/](https://github.com/coder-xiaotian/swc-useclient/blob/5802171d5d89874caad96af8ee1d6a58cb5b2531/src/lib.rs#L26-L31)

It should be noted that you need to set lto=false in Cargo.toml.
